src/widgets/menu.rs

```diff
@@ -0,0 +1,180 @@
+        }
+    }
+
+    pub fn attach<T: ::WidgetTrait>(&self, child: &T, left_attach: u32, right_attach: u32, top_attach: u32, bottom_attach: u32) {
+        unsafe {
+            ffi::gtk_menu_attach(GTK_MENU(self.unwrap_widget()), child.unwrap_widget(), left_attach, right_attach, top_attach, bottom_attach);
+        }
+    }
+
+    //pub fn attach_to_widget<T: ::WidgetTrait>(&self, attach_widget: &T, detacher: /*Unknown kind*/Unknown rust type: "MenuDetachFunc") {
+    //    unsafe { TODO: call ffi:gtk_menu_attach_to_widget() }
+    //}
+
+    pub fn detach(&self) {
+        unsafe {
+            ffi::gtk_menu_detach(GTK_MENU(self.unwrap_widget()));
+        }
+    }
+
+    // pub fn get_accel_group(&self) -> Option<AccelGroup> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_accel_group(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    pub fn get_accel_path(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gtk_menu_get_accel_path(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    // pub fn get_active<T: ::WidgetTrait>(&self) -> Option<T> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_active(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    // pub fn get_attach_widget<T: ::WidgetTrait>(&self) -> Option<T> {
+    //     unsafe {
+    //         from_glib_none(ffi::gtk_menu_get_attach_widget(GTK_MENU(self.unwrap_widget())))
+    //     }
+    // }
+
+    pub fn get_monitor(&self) -> i32 {
+        unsafe {
+            ffi::gtk_menu_get_monitor(GTK_MENU(self.unwrap_widget()))
+        }
+    }
+
+    pub fn get_reserve_toggle_size(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_menu_get_reserve_toggle_size(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn get_tearoff_state(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_menu_get_tearoff_state(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn get_title(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::gtk_menu_get_title(GTK_MENU(self.unwrap_widget())))
+        }
+    }
+
+    pub fn popdown(&self) {
+        unsafe {
+            ffi::gtk_menu_popdown(GTK_MENU(self.unwrap_widget()));
+        }
+    }
+
+    pub fn popup<T: ::WidgetTrait, U: ::WidgetTrait>(&self, parent_menu_shell: Option<&T>, parent_menu_item: Option<&U>, button: u32, activate_time: u32) {
+        unsafe {
+            let parent_menu_shell = match parent_menu_shell {
+                Some(ref p) => p.unwrap_widget(),
+                None => ::std::ptr::null_mut()
+            };
+            let parent_menu_item = match parent_menu_item {
+                Some(ref p) => p.unwrap_widget(),
+                None => ::std::ptr::null_mut()
+            };
+            ffi::gtk_menu_popup(
+                GTK_MENU(self.unwrap_widget()),
+                parent_menu_shell,
```

Comment:
Matches are too cumbersome for this, after importing `std::ptr` handling the optional comes down to:
```
parent_menu_shell.map(|p| p.unwrap_widget()).unwrap_or(ptr::null_mut())
```
which will be much easier to convert to `.to_glib_none().0` when the time comes.